ehrql/backends/tpp.py

```diff
@@ -126,7 +126,7 @@ def modify_query_variables(self, variables):
                 )
             ),
         )
-        return variables
+        return dataclasses.replace(dataset, population=new_population)
```

Comment:
I've not seen `.replace()` before.  Given `Dataset` only has two fields, it feels a bit heavy-handed.